internal/controller/processor.go

```diff
@@ -70,6 +70,9 @@ func (processor *AppConfigurationProviderProcessor) processFullReconciliation()
 		return err
 	}
 	processor.Settings = updatedSettings
+	for _, reference := range updatedSettings.SecretReferences {
+		reference.UpdateNeeded = true
+	}
 	processor.ReconciliationState.ExistingSecretReferences = updatedSettings.SecretReferences
```

Comment:
Since the SecretReferences now involves the information of secret now, we should set it to ExistingSecretReferences in the `Finish()` step now, otherwise it might mismatch with the actual version in the K8s secret.